sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/implementation/SparkBridgeImplementationInternal.scala

```diff
@@ -192,6 +195,39 @@ private[cosmos] object SparkBridgeImplementationInternal extends BasicLoggingTra
     rangeToNormalizedRange(feedRange.getEffectiveRange(pkDefinition))
   }
 
+  private[cosmos] def partitionKeyValuesToNormalizedRange
+  (
+      partitionKeyValue: Object,
```

Comment:
Why is this an object? Sounds like the expectation is that whaetverPassedHere toString retruns valid json - which is not happening in general. So, this API contract should be tightened - either by making the caller do the transformation to json and expect a json string here by narrowing down the types that can be used to ensure their toString representation is always json

Reply:
Ok

Reply:
I believe this has to be an object or string from the UDF.

Reply:
See this for an example - https://stackoverflow.com/questions/59026439/how-to-create-a-spark-udf-in-java-which-accepts-array-of-strings